File: leetcode/01614_maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        maxval = 0
        count = 0
        for i in s:
            if i == "(":
                count += 1
            elif i == ")":
                maxval = max(maxval, count)
                count -= 1

        return maxval

File: leetcode/01614_maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.js
/**
 * @param {string} s
 * @return {number}
 */
var maxDepth = function (s) {
    let maxVal = 0,
        count = 0;
    for (let i = 0; i < s.length; i++) {
        if (s[i] == "(") {
            count++;
        } else if (s[i] == ")") {
            if (count > maxVal) {
                maxVal = count;
            }
            count--;
        }
    }
    return maxVal;
};